### ISSUE TITLE
Can't connect to hitBTC ws API

### ISSUE THREAD

**Issue by jonathonmcmurray** [jonathonmcmurray/ws.q#1]
Connection fails, with a 400 Bad Request response:

```
q)\l ws.q
q).ws.VERBOSE:1b                                                //enable ws.q verbose mode
q).hbc.upd:{.hbc.x,:enlist x}                                   //define upd func for hbc
q).hbc.h:.ws.open["wss://api.hitbtc.com/api/2/ws";`.hbc.upd]    //send request
-- REQUEST --
:wss://api.hitbtc.com GET /api/2/ws HTTP/1.1
Host: api.hitbtc.com
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Version: 13
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Origin: api.hitbtc.com


-- RESPONSE --
HTTP/1.1 400 Bad Request
Server: nginx
Date: Wed, 23 May 2018 23:02:53 GMT
Content-Type: text/html
Content-Length: 11
Connection: keep-alive



```

Reported by @QuickSwitch

**Comment by jonathonmcmurray** [comment]
Issue was that when opening a WebSocket connection, kdb+ already adds all the necessary headers (`Sec-WebSocket-Key`, `Sec-WebSocket-Version`, `Sec-WebSocket-Extensions`, `Upgrade`, `Connection`) so my code was duplicating (some of) these, which led to "bad requests".

In other words, the solution is simply to simplify `ws.q` much further. With this change:

```
q).hbc.upd:{show .j.k x}
q).hbc.h:.ws.open["wss://api.hitbtc.com/api/2/ws";`.hbc.upd]
q).hbc.h
-3i
q).hbc.h .j.j `method`params`id!(`getCurrency;(1#`currency)!1#`ETH;123)
q)jsonrpc| "2.0"
result | `id`fullName`crypto`payinEnabled`payinPaymentId`payinConfirmations`payoutEnabled`payoutIsPaymentId`transferEnabled`delisted`payoutFee!("ETH";"Ethereum";1b;1b;0b;2f;1b;0b;1b;0b;"0.00958")
id     | 123f
```

Small disadvantage is that the verbose mode I added is less effective, as it won't show the kdb+ added headers on the request.

Will push a commit now that fixes this

**Comment by jonathonmcmurray** [comment]
fixed in fd655f1
